Updated packages in Python environment

emiliom opened the pull request:
Tried to update conda-lock but ran into problem with pip dependencies

abkfenris commented:
Lets just go a cross post this: https://github.com/oceanhackweek/jupyter-image/issues/42#issuecomment-1428193354

> @emiliom It looks like echopy is a simple enough package to quickly get on Conda-Forge that [grayskull](https://github.com/conda/grayskull#introduction) probably can auto scaffold the recipe for staged-recipies, so that might be the best route currently, as I don't think conda can directly ingest the non-rendered lockfiles.
>
> @ocefpaf only mamba and micromamba can use the intermediate lockfiles right?

I don't think I'm going to be able convert to a mamba/micromamba based image before this event, and if there are any issues the cross-language debugging might be messy so it's probably better not to try.

ocefpaf commented on py-base/environment.yml:
```diff
@@ -94,3 +102,8 @@ dependencies:
+    # PyPI package is older, 
+    # but installing from github leads to conda-lock error
+    # - git+https://github.com/open-ocean-sounding/echopy.git
+    - echopy
```
This package does not install its dependencies and pip installing it will bring a broken pkg, or not fully functioning at best. I sent https://github.com/conda-forge/staged-recipes/pull/22020 PR to help out a bit with its installation. It is only missing one optional dependency as far as I can tell.

emiliom replied:
Thank you!!!

emiliom commented on py-base/environment.yml:
```diff
@@ -94,3 +102,8 @@ dependencies:
   - xlrd
   - zarr
+  - pip:
+    # PyPI package is older, 
```
Thanks!

emiliom commented:
How about if I remove `echopy` from this PR? That way it can move forward with the other updates, and come back with `echopy` later, when there might also be other package additions/updates.

emiliom commented:
@abkfenris and @ocefpaf The current Python image includes `xlrd`. IMHO it is useful to have Excel read capability. But xlrd is no longer maintained and doesn't read `.xlsx` files. Do you have any concerns about replacing it with `openpyxl`?

abkfenris commented:
I don't have an issue swapping it out, especially since [Pandas supports using it](https://pandas.pydata.org/docs/user_guide/io.html#excel-files).

emiliom commented:
Thanks. 

I forgot that openpyxl doesn't support reading `.xls` files, but xlrd does. I'm inclined to still drop xlrd altogether, given that it's getting old (though it doesn't seem to have any dependencies). So, I'll do that.

ocefpaf commented:
> Do you have any concerns about replacing it with `openpyxl`?
I don't use `xlrd` anymore and we rarelly have non xlsx files. So I'm ok dropping it.

PS: I would also drop the conda-lock.yml file here b/c we only create the Linux lock anyway. You can do it with `conda-lock-f environment.yml -p linux-64`

ocefpaf commented on py-base/environment.yml:
```diff
@@ -79,18 +86,26 @@ dependencies:
+  # - xlrd  # deprecated, use openpyxl instead
   - zarr
+  # - pip:
+    # - echopy
```
This is in conda now. You can try it out here.

emiliom replied:
Thank you!!!

emiliom commented:
I've removed `conda-lock.yml` and updated the PR.

Regarding `conda-lock-f environment.yml -p linux-64`, I get an error "conda-lock-f: command not found". Maybe you meant `conda-lock -f environment.yml -p linux-64`? That's the statement I've used, per @abkfenris 's instructions in #42, followed by `conda-lock render -p linux-64`

emiliom commented:
Alright, I've re-created the conda environment (again with mamba) now with the new conda-forge `echopy` package, and updated the conda-lock file. This is now ready to go, hopefully!

emiliom commented:
Thanks!
